Nil reference in event recorder

```
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                    │
│ [signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x148e069]                                                                                                   │
│                                                                                                                                                                            │
│ goroutine 336 [running]:                                                                                                                                                   │
│ github.com/fluxcd/pkg/runtime/events.(*Recorder).AnnotatedEventf(0x0, {0x1bef760, 0xc000609b00}, 0xc0006b2720, {0x1995f5e, 0x7}, {0x19a1f9c, 0x2}, {0xc0007c8e00, 0x1ba},  │
│     /go/pkg/mod/github.com/fluxcd/pkg/runtime@v0.13.2/events/recorder.go:124 +0x69                                                                                         │
│ github.com/weaveworks/tf-controller/controllers.(*TerraformReconciler).fireEvent(_, {_, _}, {{{0x163a1d9, 0x9}, {0xc000987600, 0x20}}, {{0xc0006f1ec0, 0xa}, {0x0, ...}, . │
│     /workspace/controllers/terraform_controller.go:1436 +0x258                                                                                                             │
│ github.com/weaveworks/tf-controller/controllers.(*TerraformReconciler).Reconcile(0xc0006ce000, {0x1c06628, 0xc00098d470}, {{{0xc0006f1ed0, 0x18ca8a0}, {0xc0006f1ec0, 0x30 │
│     /workspace/controllers/terraform_controller.go:281 +0x1875                                                                                                             │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc000118580, {0x1c06628, 0xc00098d3e0}, {{{0xc0006f1ed0, 0x18ca8a0}, {0xc0006f1ec0, 0x413c │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:114 +0x26f                                                                    │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000118580, {0x1c06580, 0xc000562d00}, {0x17ea780, 0xc000988180})                  │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:311 +0x33e                                                                    │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000118580, {0x1c06580, 0xc000562d00})                                          │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:266 +0x205                                                                    │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()                                                                                       │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:227 +0x85                                                                     │
│ created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2                                                                                │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:223 +0x357       
```

--- a/cmd/manager/main.go
+++ b/cmd/manager/main.go
@@ -134,13 +134,9 @@ func main() {
 	}
 
 	var eventRecorder *events.Recorder
-	if eventsAddr != "" {
-		if er, err := events.NewRecorder(mgr, ctrl.Log, eventsAddr, controllerName); err != nil {
-			setupLog.Error(err, "unable to create event recorder")
-			os.Exit(1)
-		} else {
-			eventRecorder = er
-		}
+	if eventRecorder, err = events.NewRecorder(mgr, ctrl.Log, eventsAddr, controllerName); err != nil {
+		setupLog.Error(err, "unable to create event recorder")
+		os.Exit(1)
 	}
 
 	signalHandlerContext := ctrl.SetupSignalHandler()

